File: mobile/lib/auth/FarmerSignUpRequest.dart
class FarmerSignupRequest {
  final String username;
  final String email;
  final String password;
  final String phoneNumber;
  final String farmName;
  final double farmLocationLat;
  final double farmLocationLon;
  final double farmSize;
  final List<CropDto> farmCrops;

  FarmerSignupRequest({
    required this.username,
    required this.email,
    required this.password,
    required this.phoneNumber,
    required this.farmName,
    required this.farmLocationLat,
    required this.farmLocationLon,
    required this.farmSize,
    required this.farmCrops,
  });

  Map<String, dynamic> toJson() {
    return {
      'username': username,
      'email': email,
      'password': password,
      'phoneNumber': phoneNumber,
      'farmName': farmName,
      'farmLocationLat': farmLocationLat,
      'farmLocationLon': farmLocationLon,
      'farmSize': farmSize,
      'farmCrops': farmCrops.map((crop) => crop.toJson()).toList(),
    };
  }
}

class CropDto {
  final String cropName;
  final double area;

  CropDto({
    required this.cropName,
    required this.area,
  });

  Map<String, dynamic> toJson() {
    return {
      'cropName': cropName,
      'area': area,
    };
  }
}

File: src/main/java/com/jgroup/farmers_market/model/api/FarmerSignupRequest.java
package com.jgroup.farmers_market.model.api;

import com.jgroup.farmers_market.model.dto.CropDto;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Set;

@AllArgsConstructor
@NoArgsConstructor
@Data
public class FarmerSignupRequest {
    private String username;
    private String email;
    private String password;
    private String phoneNumber;
    private String farmName;
    private Double farmLocationLat;
    private Double farmLocationLon;
    private Double farmSize;
    private Set<CropDto> farmCrops;
}
